### DISCUSSION TITLE
a filter-regexp for topic/data/* roles

### DISCUSSION THREAD

**Discussion by baganokodo** [tchiotludo/akhq#1038] [Ideas]
Hi team,

The "topics-filter-regexp" applies to topic listing. We want everyone to be able to list all the topics, while limit users' access to the messages on different topics. Can we support a "topic-data-filter-regexp" which restricts user's access to messages on different topics?

Thank you

**Comment by tchiotludo** [comment]
hi @baganokodo, 

Can you explain me better the behaviour please ? 
- `topic-data-filter-regexp` will apply on what ? key, value, headers ? 
- on every topic on only a subset ? 
... 

Thanks

**Comment by baganokodo** [comment]
Hi [tchiotludo](https://github.com/tchiotludo),

Thank you for your timely response. The requirement is to protect messages on topics from arbitrary access. Only authorized persons can read messages on specific topics, while everyone should be able to list all the topics and inspect their configuration metadata.

I made a control plane exposing a REST endpoint for AKHQ's to connect. Upon receiving a request with a username and groups, the REST endpoint returns roles and filters as described at https://akhq.io/docs/configuration/authentifications/external.html. The "topicsFilterRegexp" filter narrows down the list of topics displaying in AKHQ, while I want a similar filter specifying the list of topics with messages viewable to the user, aligned with the "topic/data/read" role. In other words, the proposed filter specifies the topics where a user is granted to view messages in AKQH.

Hope it makes sense

Thank you

**Comment by tchiotludo** [comment]
Ok understood, 
It's cover are https://github.com/tchiotludo/akhq/pull/964 as I understand. 
There is still work but definitely on the roadmap 😉
